pet-finder/src/Components/Forms/RegistrationForm/index.tsx

```diff
@@ -1,9 +1,90 @@
-import { FC } from 'react'
+import { FC, useCallback } from 'react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+import { StyledInput } from '../../StyledInput'
+import { StyledButton } from '../../StyledButton'
+
+import { ERROR_MESSAGES } from '../../../constants/errorMessages'
+import {
+  RegisterFormType,
+  registerSchema,
+} from '../../../utils/registerFormSchema'
+
+import { logIn, signUp } from '../../../requests/req'
+import {
+  IRequestBodyLogIn,
+  IResponseBodyLogIn,
+} from '../../../requests/interfaces'
+import {
+  ExceptionMessage,
+  exceptionResponse,
+} from '../../../requests/constants'
 
 import styles from './RegistrationForm.module.scss'
+import { registerFormFields } from '../../../constants/formFields'
 
 interface RegistrationFormProps {}
 
-export const RegistrationForm: FC<RegistrationFormProps> = () => (
-  <form className={styles.registrationForm}>RegistrationForm</form>
-)
+export const RegistrationForm: FC<RegistrationFormProps> = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    setError,
+  } = useForm({
+    mode: 'all',
+    resolver: yupResolver(registerSchema(ERROR_MESSAGES)),
+  })
+
+  const handleServerError = useCallback(
+    (status: number) => {
+      const serverMessage = exceptionResponse[status as keyof ExceptionMessage]
+      setError('root.serverError', { message: serverMessage })
+    },
+    [setError]
+  )
+
+  const onSubmit: SubmitHandler<RegisterFormType> = useCallback(
+    async (requestBody) => {
+      try {
+        const singUpResponse = await signUp(requestBody)
+        if (!singUpResponse.ok) {
+          handleServerError(singUpResponse.status)
+        } else {
+          const reqBodyLogIn: IRequestBodyLogIn = {
+            email: requestBody.email,
+            password: requestBody.password,
+          }
+          const logInResponse = await logIn(reqBodyLogIn)
+          if (logInResponse.ok) {
+            const { accessToken, refreshToken }: IResponseBodyLogIn =
+              await logInResponse.json()
+            console.log(accessToken, refreshToken)
```

Comment:
console

Reply:
Lets leave this log, work in not finished here